homeassistant/components/sensor/reddit.py

```diff
@@ -0,0 +1,118 @@
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, CONF_MAXIMUM)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['praw==6.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONF_SUBREDDITS = 'subreddits'
+
+ATTR_ID = 'id'
+ATTR_BODY = 'body'
+ATTR_COMMENTS_NUMBER = 'comms_num'
+ATTR_CREATED = 'created'
+ATTR_POSTS = 'posts'
+ATTR_SUBREDDIT = 'subreddit'
+ATTR_SCORE = 'score'
+ATTR_TITLE = 'title'
+ATTR_URL = 'url'
+
+DEFAULT_NAME = 'Reddit'
+
+SCAN_INTERVAL = timedelta(seconds=300)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CLIENT_ID): cv.string,
+    vol.Required(CONF_CLIENT_SECRET): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_SUBREDDITS): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_MAXIMUM, default=10): cv.positive_int
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Reddit sensor platform."""
+    import praw
+
+    subreddits = config[CONF_SUBREDDITS]
+    user_agent = config.get(CONF_USERNAME) + '_home_assistant_sensor'
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.

Please use new style string formatting instead of string concatenation.